db.go

```diff
@@ -0,0 +1,133 @@
+package main
+
+import (
+	"fmt"
+	"github.com/jinzhu/gorm"
+	_ "github.com/jinzhu/gorm/dialects/postgres"
+	"log"
+)
+
+type CertDetail struct {
+	gorm.Model
+	Domain string `gorm:"unique"`
+	Value  string
+}
+
+type DomainAltname struct {
+	gorm.Model
+	Domain string `gorm:"unique"`
+	Value  string
+}
+
+type UserInfo struct {
+	gorm.Model
+	Email string `gorm:"unique"`
+	Value string
+}
+
+func db(dbArgs string) (*gorm.DB, error) {
+	db, err := gorm.Open("postgres", dbArgs)
+	if err != nil {
+		fmt.Println(err)
+		panic("failed to connect to certManager database")
+	}
+
+	return db, err
+}
+
+/**
+Get Domain AltNames via domain
+*/
+func getAltNames(domain string, d *gorm.DB) (altNamesRaw []byte, err error) {
+	log.Printf("Retreving domain alt-names from database for domain (%s)", domain)
+
+	var altNames DomainAltname
+	d.Where(&DomainAltname{Domain: domain}).First(&altNames)
```

Comment:
You should check for DB errors here
```suggestion
	err = d.Where(&DomainAltname{Domain: domain}).First(&altNames).Error
	if err != nil {
	    return nil, err
	}
```

Reply:
Question about the err check. Is it not okay to pass the error up to the processor and have it log it?
`userInfoRaw, err = getUserInfo(email, p.db)

	if err != nil {
		return p.NoteCertError(cert, err, "Error while running database view user information transaction for domain %v", cert.Spec.Domain)
	}`
In this case the getUserInfo is called which could contain an error but it will be part of the p.NoteCertError?

Reply:
There's nothing wrong with passing the error up, that's a typical way of handling errors in go.  If you need to add some extra context information to the error, `errors.Wrap` or `errors.Wrapf` is often used.  My point was that the code wasn't checking gorm for errors, it was always returning nil for the error.

Reply:
Sounds good. I am going to wrap the gorm error into the reg err.